tests/test_modeling_common.py

```diff
@@ -4270,6 +4270,18 @@ def test_flash_attn_2_generate_use_cache(self):
                     use_cache=True,
                 )
 
+                # Generate with one batch only to test generation when attention mask will be None
+                # when real inputs are used, because there is no padding. See issue #32237 for more
+                dummy_input = dummy_input[:1, ...]
+                dummy_attention_mask = torch.ones_like(dummy_attention_mask[:1, ...])
```

Comment:
the attention mask gets thrown away in the `update_causal_mask` function, if it's full of ones. Thus we always end up int the case were we have position ids automatically generated, and `position_ids[:, -1] == position_ids.size(1) - 1).all()`.

I should have asked for a test with dummy inputs being packed: batch size one but position ids are gonna be:
`0, 1, 2, 3, 4, 5, 6, 0, 1, 2, 3, 4, 5 ,6 ,7, 0,1,2,3]` . Which I don't think was tested

Reply:
Yes, that was the issue. I added test following the path how the issue was discovered, but surely can be tested within the `test_padding_flash_attention_with_position_ids`

Reply:
OUps, I meant can we update the test to make sure we try a case with packed inputs 😅

Reply:
No worries

Reply:
Oh i see, anyway we have to take some time to sort out tests, including flash-attn, Too many are failing now and who know when it started

Reply:
yep